packages/graph/hash_graph/lib/graph/src/store/query/mod.rs

```diff
@@ -68,10 +85,16 @@ fn compare(lhs: &Literal, rhs: &Literal) -> Result<bool, ExpressionError> {
         (lhs, Literal::List(rhs)) => rhs.iter().try_find(|rhs| compare(lhs, rhs))?.is_some(),
 
         // Version
-        (Literal::Version(lhs, _), Literal::Float(rhs)) => *lhs == *rhs as u32,
-        (Literal::Float(lhs), Literal::Version(rhs, _)) => *lhs as u32 == *rhs,
-        (Literal::String(lhs), Literal::Version(_, latest)) if lhs == "latest" => *latest,
+        (Literal::Version(Version::Ontology(lhs), _), Literal::Float(rhs)) => *lhs == *rhs as u32,
+        (Literal::Version(Version::Entity(lhs), _), Literal::Float(rhs)) => {
+            lhs.timestamp() == *rhs as i64
```

Comment:
So we need to give the timestamp of the entity back as a unix timestamp? Asking because I believe the API returns the ID back as an ISO string, i.e. ` "version": "2022-09-05T16:13:38.313197Z"` right now. It's obviously not hard not get the timestamp from this, but just flagging it's different :)

Reply:
Good point, I added string-based versions in 7a2bc2bebb98d05400c2436f3a781cb87a14fb42